aggregation/src/main/java/fk/prof/aggregation/state/AggregationState.java

```diff
@@ -0,0 +1,81 @@
+package fk.prof.aggregation.state;
+
+
+//TODO: Add ascii-art here explaining the state machine with transitions
+public enum AggregationState {
+  SCHEDULED {
+    @Override
+    public AggregationState process(AggregationStateEvent stateEvent) {
+      switch (stateEvent) {
+        case START_PROFILE:
+          return ONGOING;
+        default:
+          return this;
```

Comment:
I had deliberately not thrown exception here. process method applies the event to self and returns the new state. exception can be inferred by an outside entity which was expecting state to change once some aggregationstateevent is triggered (which we do and throw aggregation failure)